### ISSUE TITLE
Broken code generation

### ISSUE THREAD

**Issue by dowski** [bumptech/palm#31]
#29 has caused a regression. You can no longer reference messages imported from other `.proto` files.

**Comment by dowski** [comment]
Updated tests to demonstrate the issue in 42910be24df7c519b7b37252328ace63861a2ff8. Fails before the tests can even run due to a `NameError`.

```
$ nosetests test/test.py
E
======================================================================
ERROR: Failure: NameError (global name 'TYPE_Bar' is not defined)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/christian/src/bump/env/local/lib/python2.7/site-packages/nose/loader.py", line 390, in loadTestsFromName
    addr.filename, addr.module)
  File "/home/christian/src/bump/env/local/lib/python2.7/site-packages/nose/importer.py", line 39, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/home/christian/src/bump/env/local/lib/python2.7/site-packages/nose/importer.py", line 86, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/christian/src/bump/server/contrib/palm/test/test.py", line 23, in <module>
    import build.test_palm as test_palm
  File "/home/christian/src/bump/server/contrib/palm/test/build/test_palm.py", line 3981, in <module>
    eval(cname)._pbf_finalize()
  File "/home/christian/src/bump/server/contrib/palm/test/build/test_palm.py", line 194, in _pbf_finalize
    c(cls)
  File "/home/christian/src/bump/server/contrib/palm/test/build/test_palm.py", line 3966, in _finalize_ext
    if is_string(TYPE_Bar):
NameError: global name 'TYPE_Bar' is not defined

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (errors=1)
```

**Comment by dowski** [comment]
Fixed in 8b8e22540c77bcc180f890f30748326138ae8743.

**Comment by dowski** [comment]
@m4dc4p - mind taking a look at the above fix. What do you think?

A less brute force approach might be something like:

``` diff
diff --git a/palm/palmc/codegen.py b/palm/palmc/codegen.py
index 089cde1..df2cb0f 100644
--- a/palm/palmc/codegen.py
+++ b/palm/palmc/codegen.py
@@ -68,8 +68,10 @@ def gen_module(messages, imports, tlenums, with_slots, packages, curr_packa
     out('from palm.palm import ProtoBase, is_string, RepeatedSequence, ProtoValueError\n\n_PB
     for i in imports:
         converted_name = convert_proto_name(i)
-        out('import %s\n' % converted_name)
-        out('from %s import *\n' % converted_name)
+        if packages:
+            out('import %s\n' % converted_name)
+        else:
+            out('from %s import *\n' % converted_name)

     for ename, espec in tlenums:
         write_enum(ename, espec)
```

Not sure if that works when you are mixing .protos with and without package declarations. Maybe that's not allowed anyhow.
